### PR TITLE
Improve copy and assign operation

### PR DESCRIPTION
**Pull request by okuta**
`__getitem__` method has some overhead. And, It finally calls `elemntwise_copy`.

### CODE REVIEW

**Review by toslunar — COMMENTED** [review]
`dst[...] = src` and `cupy.copyto(dst, src, casting='unsafe')` utilize `dst.data.copy_from_async` if memcpy is possible.  I suppose a faster `cupy.copyto` (without shape/casting checks) is what we need.

**Comment by okuta** [comment]
OK, I will rewrite.

**Comment by okuta** [comment]
I checked `copyto`.  Basically, copying for `out` does not cause non-contiguous cases. So I think there is no problem with `elementwise_copy`.

What do you think?

File: cupyx/scipy/ndimage/morphology.py [hunk 1]
```diff
@@ -257,7 +258,7 @@ def _binary_erosion(input, structure, iterations, mask, output, border_value,
                 break
         output = tmp_out
     if temp_needed:
-        temp[...] = output
+        _core.elemntwise_copy(output, temp)
```
Comment (grlee77) [inline cupyx/scipy/ndimage/morphology.py:261]:
```suggestion
        _core.elementwise_copy(output, temp)
```

Comment (okuta) [reply cupyx/scipy/ndimage/morphology.py:261]: Thanks

**Review by takagi — APPROVED** [review]

**Pull request merged**